### PR TITLE
Do not create billing records for the VMs in internal projects

### PR DESCRIPTION
**Pull request by enescakir**
Our internal projects generate thousands of billing records for GitHub runners. However, these aren't particularly useful since we maintain a 'deleted_record' table for auditing. We decided to refrain from creating billing records for our internal projects until we implement a data warehouse solution.

I moved the creation of billing records to a separate label, after 'wait_sshable'. This ensures that customers are not charged if the VM is stuck in 'wait_sshable'. Additionally, having an individual label similar to the postgres resource has enhanced clarity.

### CODE REVIEW

File: prog/vm/nexus.rb [hunk 1]
```diff
@@ -293,8 +293,26 @@ def before_run
...
+      vm.update(display_state: "running")
+      Clog.emit("vm provisioned") { {vm: vm.values, provision: {vm_ubid: vm.ubid, vm_host_ubid: host.ubid, duration: Time.now - vm.created_at}} }
+      hop_create_billing_record
+    end
+    nap 1
+  end
+
+  label def create_billing_record
+    project = vm.projects.first
+    hop_wait unless project.billable
```
Comment (byucesoy) [inline prog/vm/nexus.rb:312]: I think we can move this check to wait_sshable (i.e. we would directly hop to wait skipping create_billing_record)

**Review by byucesoy — APPROVED** [review]

File: prog/vm/nexus.rb (hunk 1 above)
Comment (enescakir) [reply prog/vm/nexus.rb:312]: I believe that checking it as the first step of the skipped label enhances clarity and maintainability.

https://github.com/ubicloud/ubicloud/pull/96#issuecomment-1524732612

Comment (byucesoy) [reply prog/vm/nexus.rb:312]: fair enough. I'm not sure which way is better, but I don't have strong preference either.

**Comment by enescakir** [comment]
@fdr What do you think about it?

**Review by fdr — APPROVED** [review]
I give my subordinate approval (it's probably not as accurate as others that have reviewed)

**Review by furkansahin — APPROVED** [review]

**Pull request merged**